### ISSUE TITLE
Support conditionals

### ISSUE THREAD

**Issue by fsteeg** [metafacture/metafacture-fix#10]
For a Fix like:

```
if equals(html.head.meta.property, 'description')
    map(html.head.meta.content, 'description')
```

With input like:

```
<meta property="description" content="Das Bewusstsein und..."/>
```

See also links in _conditionals_ bullet point at https://github.com/metafacture/metafacture-fix#state

See also https://github.com/metafacture/metafacture-core/issues/312

**Comment by fsteeg** [comment]
Input data samples:

https://www.hoou.de/materials/tutorial-lernen-lernen
(`html.head.meta.content`, `html.head.meta.name`, `link.rel`)

https://duepublico2.uni-due.de/servlets/OAIDataProvider?verb=ListRecords&metadataPrefix=mods
(`metadata.mods.location.url`)

**Comment by fsteeg** [comment]
The original sample above probably needs to be combined with a new `list` bind, like:

```
do list(path: html.head.meta.*, var: meta)
  if equals(meta.property, 'description')
    map(meta.content, description)
  end
end
```

See https://github.com/LibreCat/Catmandu/wiki/Binds

**Comment by fsteeg** [comment]
While we need support for conditionals in general, the examples here might be approached in a different way, resulting in more readable mapping code. See https://github.com/metafacture/metafacture-core/issues/350

**Comment by fsteeg** [comment]
Original use case from first comment was solved with https://github.com/metafacture/metafacture-core/issues/350, basic if statements implemented in https://github.com/metafacture/metafacture-fix/pull/51, closing.
